morpheus/service/vdb/milvus_vector_db_service.py

```diff
@@ -217,6 +384,158 @@ async def similarity_search(self,
 
         return outputs
 
+    def update(self, data: list[typing.Any], **kwargs: dict[str, typing.Any]) -> dict[str, typing.Any]:
+        """
+        Update data in the collection.
+
+        Parameters
+        ----------
+        data : list[typing.Any]
+            Data to be updated in the collection.
+        **kwargs : dict[str, typing.Any]
+            Extra keyword arguments specific to upsert operation.
+
+        Returns
+        -------
+        dict[str, typing.Any]
+            Returns result of the updated operation stats.
+        """
+
+        if not isinstance(data, list):
+            raise RuntimeError("Data is not of type list.")
+
+        result = self._client.upsert(collection_name=self._name, entities=data, **kwargs)
+
+        self._collection.flush()
+
+        return self._update_delete_result_to_dict(result=result)
+
+    def delete_by_keys(self, keys: int | str | list, **kwargs: dict[str, typing.Any]) -> typing.Any:
+        """
+        Delete vectors by keys from the collection.
+
+        Parameters
+        ----------
+        keys : int | str | list
+            Primary keys to delete vectors.
+        **kwargs :  dict[str, typing.Any]
+            Extra keyword arguments specific to the vector database implementation.
+
+        Returns
+        -------
+        typing.Any
+            Returns result of the given keys that are deleted from the collection.
+        """
+
+        result = self._client.delete(collection_name=self._name, pks=keys, **kwargs)
```

Comment:
Just want to be sure this is correct? Other patterns appear to do something like:

```python
self._command
self._flush
```

Reply:
Delete doesn't need flush command. They are automatically getting updated in the collection. When I do search operation on the collection after delete. It is retrieving only keys that are not deleted.

We have a tests covering this scenario:
https://github.com/nv-morpheus/Morpheus/blob/b67882a6f62c304a6eb6260fa1d228e7e5ada2dc/tests/test_milvus_vector_db_service.py#L309
https://github.com/nv-morpheus/Morpheus/blob/b67882a6f62c304a6eb6260fa1d228e7e5ada2dc/tests/test_milvus_vector_db_service.py#L333